src/main/java/org/apache/commons/lang3/math/NumberUtils.java

```diff
@@ -317,38 +317,67 @@ public static Long createLong(final String str) {
      * @return Number created from the string (or null if the input is null)
      * @throws NumberFormatException if the value cannot be converted
      */
-    public static Number createNumber(final String str) {
+    public static Number createNumber(final String str, boolean [] flags) {
+        flags[0] = true;
+
         if (str == null) {
+            flags[1] = true;
             return null;
         }
```

Comment:
need to add "else" and set corresponding flags. [According to the third point of 3.5.1: 3. Before the first statement of each branch outcome **(including to-be-added “else” clauses if none exist)**, add a line that sets a flag if the branch is reached.]

Reply:
I understand what you mean but it would not be possible to reach the following flags unless both cases were tested  because the if contains a return statement, so I feel like including an else here is unnecessary :)

Reply:
I get your points, you are right, not all "else" is needed.

Reply:
Even though you're right about other flags being unreachable it should technically have a flag for the `else` also.